Plutarch/Bool.hs

```diff
@@ -83,13 +84,20 @@ class PEq t where
 
 infix 4 #==
 
-class POrd t where
+class PEq t => POrd t where
   (#<=) :: Term s t -> Term s t -> Term s PBool
   (#<) :: Term s t -> Term s t -> Term s PBool
 
 infix 4 #<=
 infix 4 #<
 
+instance PEq PBool where
+  x #== y' = plet y' $ \y -> pif' # x # y #$ pnot # y
```

Comment:
You should always `plet` something when you use something twice. If it's a variable reference it will be eliminated as an optimisation. There are few cases where it's better not to `let` in practice.